### PR TITLE
Fix bad in-place op in _generate_unfixed_lin_constraints

### LINKED ISSUE #898: [Bug] Constrained Optimization with Fixed Features and batch size q > 1 - constraints are violated midway through the optimization loop.
# 🐛 Bug

I'm performing constrained optimization with fixed features. Here, q > 1 (e.g. 8). Somehow, the constraints are failing after the first candidate of the batch is found. After some digging, I found that some of the inequality constraints have been modified causing some of the fixed features to violate the constraints. I'm not sure where the source of the bug is.

Secondly, in a related problem, when there are multiple constraints, the fixed features constraints check:  "new_rhs > 0" is quite strict (botorch\optim\parameter_constraints.py). I find that occasionally, due to numerical issues, it violates the constraints by some small number (e.g. 5e-8). I should note that the fixed features I'm using come from parts of a candidate found by the same optimizer (with the same set of inequality constraints). Do you know how I should best try to resolve this? Note, there is no rounding happening as far as I could tell. 

# all indices were fixed, so the constraint is gone.
if len(new_indices) == 0:
    if (eq and new_rhs != 0) or (not eq and new_rhs > 0):
        prefix = "Eq" if eq else "Ineq"
        raise CandidateGenerationError(
            f"{prefix}ality constraint {constraint_id} not met "
            "with fixed_features."
        )

Also there may be a typo in the error msg ("Ineqality")

** Stack trace/error message **
File "C:\Anaconda3\envs\ai-engine-updated\lib\site-packages\botorch\optim\optimize.py", line 113, in optimize_acqf
    candidate, acq_value = optimize_acqf(
  File "C:\Anaconda3\envs\ai-engine-updated\lib\site-packages\botorch\optim\optimize.py", line 184, in optimize_acqf
    batch_candidates_curr, batch_acq_values_curr = gen_candidates_scipy(
  File "C:\Anaconda3\envs\ai-engine-updated\lib\site-packages\botorch\generation\gen.py", line 97, in gen_candidates_scipy
    _no_fixed_features = _remove_fixed_features_from_optimization(
  File "C:\Anaconda3\envs\ai-engine-updated\lib\site-packages\botorch\generation\utils.py", line 131, in _remove_fixed_features_from_optimization
    inequality_constraints = _generate_unfixed_lin_constraints(
  File "C:\Anaconda3\envs\ai-engine-updated\lib\site-packages\botorch\optim\parameter_constraints.py", line 312, in _generate_unfixed_lin_constraints
    raise CandidateGenerationError(
botorch.exceptions.errors.CandidateGenerationError: Ineqality constraint 6 not met with fixed_features.

## Expected Behavior

Be able to run constrained optimization with fixed features (that are potentially close to the boundaries of the constraints).

## System information

**Please complete the following information:**
-  0.5.1.dev23+g1d35f168
-  1.5.0
-  '1.9.0+cu111'
-  Windows 10

### PR DESCRIPTION
**Pull request by Balandat**
Fixes #898

### CODE REVIEW

**Comment by vishwakftw** [comment]
Could you also fix the spelling error a few lines later: `f"{prefix}uality` instead of `f"{prefix}ality`? Thank you.

File: botorch/optim/parameter_constraints.py [hunk 1]
```diff
@@ -303,14 +303,14 @@ def _generate_unfixed_lin_constraints(
                 new_coefficients.append(coefficient)
             # otherwise, we "remove" the constraints corresponding to that index
             else:
-                new_rhs -= coefficient.item() * ffval_or_None
+                new_rhs = new_rhs - coefficient.item() * ffval_or_None
```
Comment (vishwakftw) [inline botorch/optim/parameter_constraints.py:306]: Thank you for fixing this so quickly. I think `rhs` might have been of the wrong type in the bug report causing this.
Nonetheless, this is a harmless change.

**Review by vishwakftw — APPROVED** [review]
LGTM

File: botorch/optim/parameter_constraints.py (hunk 1 above)
Comment (Balandat) [reply botorch/optim/parameter_constraints.py:306]: Yeah I think ideally we can support both python scalars and single-element tensors.

**Pull request closed without merging**